dvc/api.py

```diff
@@ -69,3 +75,108 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, fname="dvcsummon.yaml", args=None, repo=None, rev=None):
+    # TODO: Write a meaningful docstring about `summon`
+    with _make_repo(repo, rev=rev) as _repo:
+
+        def pull_dependencies(deps):
+            if not deps:
+                return
+
+            outs = [_repo.find_out_by_relpath(dep) for dep in deps]
+
+            with _repo.state:
+                for out in outs:
+                    _repo.cloud.pull(out.get_used_cache())
+                    out.checkout()
+
+        path = os.path.join(_repo.root_dir, fname)
+        obj = _get_object_from_summoners_file(name, path)
+        info = obj["summon"]
+
+        pull_dependencies(info.get("deps"))
+
+        _args = copy.deepcopy(info.get("args", {}))
+        _args.update(args or {})
+
+        return _invoke_method(info["call"], _args, path=_repo.root_dir)
+
+
+def _get_object_from_summoners_file(name, path):
+    """
+    Given a summonable object's name, search for it on the given file
+    and bring it to life.
+    """
+    schema = Schema(
+        [
+            {
+                "name": str,
+                "description": str,
+                "paper": str,
+                "metrics": dict,
+                "summon": {
+                    "type": "python",
+                    "call": str,
+                    "args": dict,  # XXX: Optional
+                    "deps": [str],  # XXX: Optional
+                },
+            }
+        ]
+    )
+
+    with open(path, "r") as fobj:
+        try:
+            objects = ruamel.yaml.safe_load(fobj.read())["objects"]
+            objects = schema(objects)
+            return next(x for x in objects if x["name"] == name)
+        except FileNotFoundError:
+            pass  # XXX: No such YAML file with path: '<path>'
+        except ruamel.yaml.ScannerError:
+            pass  # XXX: Failed to parse YAML correctly
+        except KeyError:
+            pass  # XXX: YAML file doesn't include the "objects" keyword
+        except Invalid:
+            pass  # XXX: YAML file dosen't match with the schema
+        except StopIteration:
+            pass  # XXX: No such object with name: '<name>'
+
+
+def _invoke_method(call, args, path):
+    # XXX: Some issues with this approach:
+    #   * Not thread safe
+    #   * Import will pollute sys.modules
+    #   * Weird errors if there is a name clash within sys.modules
```

Comment:
There is no good non complicated way to handle this in Python 2. I will come up with an approach next sprint. Leave as is for now.